pkg/export/otel/metrics_test.go

```diff
@@ -511,6 +511,41 @@ func TestAppMetrics_ResourceAttributes(t *testing.T) {
 	assert.Equal(t, "upstream.beyla", attributes["source"])
 }
 
+func TestAppMetrics_TracesHostInfo(t *testing.T) {
+	defer restoreEnvAfterExecution()()
+
+	ctx := t.Context()
+
+	otlp, err := collector.Start(ctx)
+	require.NoError(t, err)
+
+	now := syncedClock{now: time.Now()}
+	timeNow = now.Now
+
+	metrics := msg.NewQueue[[]request.Span](msg.ChannelBufferLen(20))
+	otelExporter := makeExporter(ctx, t, []string{instrumentations.InstrumentationHTTP}, []string{FeatureApplication, FeatureApplicationHost}, otlp, metrics)
+
+	go otelExporter(ctx)
+
+	assert.Len(t, otlp.Records(), 0, "metric reported before the first span is sent")
+
+	metrics.Send([]request.Span{
+		{Service: svc.Attrs{UID: svc.UID{Instance: "foo"}}, Type: request.EventTypeHTTP, Path: "/foo", RequestStart: 100, End: 200},
+	})
+
+	res := readNChan(t, otlp.Records(), 2, timeout)
+	assert.Len(t, res, 2)
+
+	found := false
+	for _, r := range res {
+		if r.Name == "traces_host_info" {
+			found = true
+			break
+		}
+	}
+	assert.True(t, found, "traces_host_info metric not found in the exported metrics")
```

Comment:
This is great, do you think it's possible to refactor the function so that we can directly call it and then ensure the call to host info delete happened?